node_modules/oae-principals/lib/definitiveDeletion.js

```diff
@@ -0,0 +1,1505 @@
+                        AuthzAPI.updateRoles(elementId, update, function(err, usersToInvalidate) {
+                            if(err) {
+                                return callback(err);
+                            }
+                            return callback(null);                    
+                        });
+                    }
+                });
+            } else {
+                return callback(null);
+            }
+        }
+    });
+};
+
+
+
+
+/** =============================================================== **/
+/** ===================== Definitive deletion ===================== **/
+/** =============================================================== **/
+
+/**
+ * Definitive delete of user
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
+ * @param  {String}     user            User that will be deleted
+ * @param  {String}     alias           Tenant alias
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occured, if any
+ */
+var definitiveDeletion = module.exports.definitiveDeletion = function(ctx, user, alias, callback) {
+    callback = callback || function() {};
+
+    // Get userArchive
+    PrincipalsDAO.getUserArchive(alias, function(err, userArchive) {
+        if (err) {
+            return callback(err);
+        }   
+
+        // Delete all rights on resources
+        _deleteRightResources(ctx, user, userArchive.idArchive, function(err) {
+            if (err) {
+                return callback(err);
+            } 
+
+            // Mark the user as deleted and then, deleted him
+            PrincipalsDAO.deletePrincipal2(ctx, user, function(err) {
+                if (err) {
+                    return callback(err);
+                }
+
+                // Notify consumers that a user has been deleted
+                log().info({'userId': user.id, 'name': 'oae-principals'}, 'Definitive deletion user with a mapped login id');
+
+                // Notify that a user has been deleted
+                PrincipalsEmitter.emit(PrincipalsConstants.events.DELETED_USER, ctx, user, function(err) {
+                    if (err) {
+                        return callback(err);
+                    }
+                    return callback(null, true);
+                });
+            });
+        });
+    });
+};
+
+/**
+ * Remove all his rights on resources
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
+ * @param  {String}     user            The user to delete
+ * @param  {String}     idArchive       The user archive of the tenant
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occured, if any
+ * @api private
+ */
+var _deleteRightResources = function(ctx, user, idArchive, callback) {
```

Comment:
I think `_deleteResourceRights` would be a better name for this function, since it explains more clearly what the function does.